pkg/apis/core/types_shoot.go

```diff
@@ -1205,6 +1208,30 @@ type MaintenanceAutoUpdate struct {
 	MachineImageVersion *bool
 }
 
+// MaintenanceAutoRotation contains information about which rotations should be automatically performed.
+type MaintenanceAutoRotation struct {
+	// Credentials contains information about which credentials should be automatically rotated.
+	Credentials *MaintenanceCredentialsAutoRotation
+}
+
+// MaintenanceCredentialsAutoRotation contains information about which credentials should be automatically rotated.
+type MaintenanceCredentialsAutoRotation struct {
+	// ETCDEncryptionKey configured the automatic rotation for the etcd encryption key.
+	ETCDEncryptionKey *MaintenanceRotationConfig
+	// Observability configured the automatic rotation for the observability credentials.
+	Observability *MaintenanceRotationConfig
+	// SSHKeypair configured the automatic rotation for the ssh keypair for worker nodes.
+	SSHKeypair *MaintenanceRotationConfig
+}
+
+// MaintenanceRotationConfig contains configuration for automatic rotation.
+type MaintenanceRotationConfig struct {
+	// Enabled indicates whether automatic rotation should be performed (default: false).
```

Comment:
You could avoid specifying default value in the docstring of internal type, because defaulting happens for the external type. Theoretically, two different external types could have different default values (same for `RotationPeriod`).
```suggestion
	// Enabled indicates whether automatic rotation should be performed.
```